server-spi-private/src/main/java/org/keycloak/services/clientpolicy/ClientPoliciesUtil.java

```diff
@@ -0,0 +1,747 @@
+        List<ClientPolicyRepresentation> updatingPoliciesList = updatingPoliciesRep.getPolicies();
+
+        // add existing builtin policies to updating policies
+        List<ClientPolicyRepresentation> existingPoliciesList = existingPoliciesRep.getPolicies();
+        if (existingPoliciesList != null && !existingPoliciesList.isEmpty()) {
+            existingPoliciesList.stream().filter(i->i.isBuiltin()).forEach(i->updatingPoliciesList.add(i));
+        }
+
+        for (ClientPolicyRepresentation proposedPolicyRep : proposedPoliciesRep.getPolicies()) {
+            if (proposedPolicyRep.getName() == null) {
+                continue;
+            }
+
+            // ignore newly proposed builtin policy because builtin policy cannot added/deleted/modified.
+            Boolean enabled = (proposedPolicyRep.isEnable() != null) ? proposedPolicyRep.isEnable() : Boolean.FALSE;
+            if (proposedPolicyRep.isBuiltin() != null && proposedPolicyRep.isBuiltin()) {
+                // only enable field of the existing builtin policy can be overridden.
+                if (updatingPoliciesList.stream().anyMatch(i->i.getName().equals(proposedPolicyRep.getName()))) {
+                    updatingPoliciesList.stream().filter(i->i.getName().equals(proposedPolicyRep.getName())).forEach(i->i.setEnable(enabled));
+                    continue;
+                }
+                logger.warnv("{0} builtin policy included in proposed policies. Just ignore it. policy name = {1}", LOGMSG_PREFIX, proposedPolicyRep.getName());
+                continue;
+            }
+
+            // basically, proposed policy totally overrides existing policy except for enabled field..
+            ClientPolicyRepresentation policyRep = new ClientPolicyRepresentation();
+            policyRep.setName(proposedPolicyRep.getName());
+            policyRep.setDescription(proposedPolicyRep.getDescription());
+            policyRep.setBuiltin(Boolean.FALSE);
+            policyRep.setEnable(enabled);
+
+            policyRep.setConditions(new ArrayList<>());
+            if (proposedPolicyRep.getConditions() != null) {
+                proposedPolicyRep.getConditions().stream().filter(condition->isValidCondition(session, condition)).forEach(condition->policyRep.getConditions().add(condition));
+            }
+
+            Set<String> existingProfileNames = new HashSet<>();
+            ClientProfilesRepresentation reps = getClientProfilesRepresentation(session, realm);
+            if (reps.getProfiles() != null) {
+                reps.getProfiles().stream().map(profile->profile.getName()).forEach(profileName->existingProfileNames.add(profileName));
+            }
+            policyRep.setProfiles(new ArrayList<>());
+            if (proposedPolicyRep.getProfiles() != null) {
+                proposedPolicyRep.getProfiles().stream().filter(profileName->existingProfileNames.contains(profileName)).distinct().forEach(profileName->policyRep.getProfiles().add(profileName));
```

Comment:
Again, my vote is to throw the error in case of referencing not existing profiles

Reply:
@mposolda Yes. I would like to fix codes to throw the error as you previously proposed.